Second round with updated syntax and file distribution

FedG-code opened the pull request.

FedG-code committed:
Converting markdown files to html

Added the marked js node.
Added converter.js which converts the markdown blogs into html.
Made new directories called html-blogs and markdown-blogs to separate the files properly.

FedG-code committed:
Applied most fixes

Changed directory and file distribution.
Used cleaner string concatenation and replace methods.

Still missing: one of the files isn't converting the initial header correctly even though it looks fine in markdown and is the same as the others.  const {readdirSync} was used but likely improperly as it ended up using more code.

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,20 @@
+
+console.log(listOfMarkdownFiles);
+
+listOfMarkdownFiles.forEach(function(element){
```
you can do this in the new way with arrow functions

```
listOfMarkdownFiles.forEach(element => {
...
})
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

robinglen replied:
I would also consider something more descriptive for the variable name like `file`

FedG-code replied:
So that's a variable? You can make variables without even using var or const?

robinglen replied:
so youre not declaring a variable, its a value being return by the method `forEach`

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,20 @@
+const {readdirSync} = require('fs');
```
you can do this as one line, for each object
```
const {readdirSync, readFileSync, writeFileSync} = require('fs');
```

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,20 @@
+    const fileContent = readFileSync(`markdown/${element}`, 'utf8');
+    //Turn content to string
+   //pass the file through marked
+   var newHTMLFile = marked(fileContent);
```
this can be a `const`

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,20 @@
+   //get the new file name
+   const newFileName = element.replace('.md', '.html');
+   //write the html file with its new name
+   writeFileSync(newFileName, newHTMLFile);
```
This is stylist or you can say being "clean" but it doesn't really matter but as you're not using these variables more than once, you can do this

```
writeFileSync(element.replace('.md', '.html'), marked(fileContent));
```

Some people don't like that as what you have is easier to read... so its not critical but if youre not using a variable more than once for memory reasons should you even set it? To be honest the garbage collector solves all of this... but just something good for you to know

robinglen reviewed (commented):
some more things to make it even cleaner

FedG-code committed:
Implemented fixes as requested

FedG-code committed:
Added title formatting to the converter.js

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,20 @@
+const {readdirSync, readFileSync, writeFileSync} = require('fs');
+
+var showdown = require('showdown');
```
you can `const` this

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,20 @@
+
+var showdown = require('showdown');
+
+var convertToHTML = new showdown.Converter();
```
`const` here too

robinglen commented on blogs/package.json:
```diff
@@ -0,0 +1,17 @@
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "marked": "^4.1.0",
```
Do you want to use marked or showed? Remove the one you don't want to use from here, then do another 
```
npm i
```

robinglen commented on blogs/markdown/microservices_solving-a-problem-like-routine_2020-update.md:
```diff
@@ -1,4 +1,7 @@
-﻿# Microservices: Solving a problem like routing — 2020 update
+﻿
```
we removing these lines?

robinglen commented on blogs/converter.js:
```diff
@@ -0,0 +1,20 @@
+   //pass the file through marked
+   const newHTMLFile = convertToHTML.makeHtml(formattedFile);
+   //write the html file with its new name
+   writeFileSync(file.replace('.md', '.html'), newHTMLFile);
```
will this put it in the `blogs` directory? We should also rename that directory to 'blog'

FedG-code replied:
yeah because the code is in the blog directory

FedG-code committed:
Found the pesky character

The file is finally formatted to remove the invisible character messing with the html files

FedG-code committed:
Removed the marked module that was reintroduced for testing
